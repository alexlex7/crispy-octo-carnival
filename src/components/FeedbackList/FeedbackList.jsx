import React from 'react';
import styles from './FeedbackList.module.css';
import Container from '../Container/Container';
import PropTypes from 'prop-types';
import FeedbackItem from '../FeedbackItem/FeedbackItem';

export default function FeedbackList({ items }) {
  return (
    <Container>
      <ul className={styles.list}>
        {items.map((item, idx) => (
          <FeedbackItem key={idx} item={item} />
        ))}
      </ul>
    </Container>
  );
}

FeedbackList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      firstName: PropTypes.string.isRequired,
      feedback: PropTypes.string.isRequired,
      rating: PropTypes.number.isRequired,
    }).isRequired
  ).isRequired,
};
