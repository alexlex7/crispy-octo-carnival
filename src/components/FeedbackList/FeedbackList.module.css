.list {
  margin: 0 auto;
  display: flex;
  justify-content: center;
}
@media screen and (max-width: 1204px) {
  .list {
    margin-bottom: 50px;
  }
}
@media screen and (min-width: 1205px) {
  .list {
    padding-bottom: 140px;
    column-gap: 30px;
  }
}
