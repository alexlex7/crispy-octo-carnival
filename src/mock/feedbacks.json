[
  {
    "firstName": "Alex",
    "lastName": "Losev",
    "phone": "380986670595",
    "email": "a@t.com",
    "feedback": "The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!",
    "department": "value1",
    "permission": true,
    "rating": 5,
    "id": 7
  },
  {
    "firstName": "Alex",
    "lastName": "Losev",
    "phone": "380986670595",
    "email": "a@t.com",
    "feedback": "The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!",
    "department": "value1",
    "permission": true,
    "rating": 3.5,
    "id": 7
  },
  {
    "firstName": "Alex",
    "lastName": "Losev",
    "phone": "380986670595",
    "email": "a@t.com",
    "feedback": "The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!",
    "department": "value1",
    "permission": true,
    "rating": 4.5,
    "id": 7
  }
]
