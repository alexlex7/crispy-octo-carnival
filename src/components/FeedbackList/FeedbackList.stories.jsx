import React from 'react';
import feedbacks from '../../mock/feedbacks.json';
import FeedbackList from './FeedbackList';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Feedback/FeedbackList',
  component: FeedbackList,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <FeedbackList {...args} />;

export const Mobile = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Mobile.args = {
  items: feedbacks.slice(0, 1),
};

export const Desktop = Template.bind({});
Desktop.args = {
  items: feedbacks,
};
