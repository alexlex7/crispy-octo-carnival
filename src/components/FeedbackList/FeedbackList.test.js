import { render, screen } from '@testing-library/react';
import FeedbackList from './FeedbackList';
import feedbacks from '../../mock/feedbacks.json';

test('render feedback list component', () => {
  const { asFragment } = render(<FeedbackList items={feedbacks} />);
  const linkElement = screen.getByRole('list');
  expect(linkElement).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
