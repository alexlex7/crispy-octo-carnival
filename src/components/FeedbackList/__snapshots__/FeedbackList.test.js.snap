// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render feedback list component 1`] = `
<DocumentFragment>
  <div
    class="container "
  >
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <p
          class="name"
        >
          Alex
        </p>
        <div
          class="wrapper"
        >
          <div
            class="ratingWrapper"
          >
            <div
              class="starsWrapper"
            >
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
            </div>
            <div
              class="starsActiveWrapper"
              style="width: 100%;"
            >
              <div
                style="display: flex;"
              >
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
              </div>
            </div>
          </div>
          <p>
            5
          </p>
        </div>
        <p
          class="text"
        >
          The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!
        </p>
      </li>
      <li
        class="item"
      >
        <p
          class="name"
        >
          Alex
        </p>
        <div
          class="wrapper"
        >
          <div
            class="ratingWrapper"
          >
            <div
              class="starsWrapper"
            >
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
            </div>
            <div
              class="starsActiveWrapper"
              style="width: 70%;"
            >
              <div
                style="display: flex;"
              >
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
              </div>
            </div>
          </div>
          <p>
            3.5
          </p>
        </div>
        <p
          class="text"
        >
          The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!
        </p>
      </li>
      <li
        class="item"
      >
        <p
          class="name"
        >
          Alex
        </p>
        <div
          class="wrapper"
        >
          <div
            class="ratingWrapper"
          >
            <div
              class="starsWrapper"
            >
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
              <svg
                class="icon"
                fill="none"
              >
                star.svg
              </svg>
            </div>
            <div
              class="starsActiveWrapper"
              style="width: 90%;"
            >
              <div
                style="display: flex;"
              >
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
                <svg
                  class="activeIcon"
                  fill="orange"
                >
                  star.svg
                </svg>
              </div>
            </div>
          </div>
          <p>
            4.5
          </p>
        </div>
        <p
          class="text"
        >
          The best furniture I have seen! Thank you for yor service. The furniture has excellent quality. !!!!!!!!!!!!!!!!!!!!!!!!!!!
        </p>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
